doc/dev_guide/create_aws_sts_setup_on_minikube.md

```diff
@@ -0,0 +1,239 @@
+
+#### Requirements:
+- AWS account (with Admin permission)
+- AWS CLI
+- minikube
+- openssl
+
+### Background
+AWS STS allows applications to provide a Json Web Token (JWT) that can be used to assume an IAM role. The JWT includes an Amazon Resource Name (ARN) for `sts:AssumeRoleWithWebIdentity` IAM action to allow temporarily granted permission for the `ServiceAccount` to do this. The JWT contains the signing keys for the `ProjectedServiceAccountToken` that can be validated by AWS IAM.
+
+### Initial Setup - Create S3 Bucket (the OIDC Bucket)
+1. Create the AWS S3 bucket (for hosting the OIDC configurations). Please make it public access and ACL enabled after creation (see notes below).
+
+```bash
+aws s3api create-bucket --bucket <bucket_name> --region <aws_region> --create-bucket-configuration LocationConstraint=<aws_region>
+```
+
+#### Notes:
+- If the value of <aws_region> is us-east-1, do not specify the LocationConstraint parameter.
+- We will refer to this bucket name as <oidc_bucket_name> in the next instructions.
+- Change the publicity: in the AWS Console: block public access (bucket settings), do not block anything, and enable the bucket with ACL.
+- Can be installed using AWS Console UI: S3 &rarr; Buckets &rarr; Create Bucket &rarr; Fill the bucket name + region &rarr;  ACLs enabled &rarr; Remove the V from Block all public access &rarr; Use the rest of the defaults.
+
+2. Save the bucket name as a variable (replace `<oidc_bucket_name>` with the name that you chose for the bucket):
+
+```bash
+OPENID_BUCKET_NAME='<oidc_bucket_name>'
+```
+
+### General Setup:
+3. Create a directory for files to save the files in one-place, for example: `~/Documents/my_sts` and make sure that it is empty.
+
+```bash
+mkdir -p ~/Documents/my_sts
+```
+
+```bash
+cd ~/Documents/my_sts
+```
+
+```bash
+ls -al
+```
+
+4. Save this variable (on every terminal that we use):
+
+```bash
+OPENID_BUCKET_URL='https://<oidc_bucket_name>.s3.<aws_region>.amazonaws.com'
+```
+
+`<oidc_bucket_name>` is the bucket you created in the initial step and the `<aws_region>` is the region of this bucket (replace them with your details).
+
+### Minikube Setup:
+5. Start Minikube with the following flags:
+
+```bash
+minikube start --extra-config=apiserver.service-account-issuer=${OPENID_BUCKET_URL} --kubernetes-version=v1.23.12
+```
+
+I added the flag of `--kubernetes-version=v1.23.12` because iI had errors of timeout without it.
+
+6. After the minikube node has started, fetch the Service Account signing public key:
+
+```bash
+cd ~/Documents/my_sts 
+```
+
+```bash
+minikube ssh sudo cat /var/lib/minikube/certs/sa.pub > sa-signer.pub
+```
+
+You'll be able to see the new created file  `sa-signer.pub`.
+
+```bash
+ls -al
+```
+
+Note: every time you do `minikube delete` and `minikube start` you will need to create this new file (and the 2 files that are described in next steps).
+
+### Creating AWS resources manually - Using the Amazon web console:
```

Comment:
Can you also add the aws cli steps?

Reply:
I don't have them we only used the AWS Console and they are quite complicated (for example 'get thumbprint'). Also, the commands that are on the documentation create them differently (for example the audience is not 'api'). If you have them you can suggest and I'll be glad to attach them.

Reply:
ok, for now we can have only AWS console instructions.